source/opt/debug_info_manager.cpp

```diff
@@ -99,6 +99,8 @@ void DebugInfoManager::RegisterDbgFunction(Instruction* inst) {
   assert(inst->GetOpenCL100DebugOpcode() == OpenCLDebugInfo100DebugFunction &&
          "inst is not a DebugFunction");
   auto fn_id = inst->GetSingleWordOperand(kDebugFunctionOperandFunctionIndex);
+  // Do not register function that has been optimized away
+  if (fn_id == GetDebugInfoNone()->result_id()) return;
```

Comment:
Since we can add another `DebugInfoNone` and do not reuse it for multiple `DebugFunction`, it can be different from `GetDebugInfoNone()->result_id()` even when it is actually `DebugInfoNone`.

Why don't we change this to
```
// Skip if Function operand of DebugFunction is DebugInfoNone
if (GetDbgInst(fn_id) != nullptr) {
  assert(GetDbgInst(fn_id)->GetOpenCL100DebugOpcode() == OpenCLDebugInfo100DebugInfoNone);
  return;
}
```
?

Reply:
Change made and test added. I believe I have addressed all concerns.